diff -x '*~' -urN xorg-server-1.11.4.orig/hw/xfree86/modes/xf86EdidModes.c xorg-server-1.11.4/hw/xfree86/modes/xf86EdidModes.c
--- xorg-server-1.11.4.orig/hw/xfree86/modes/xf86EdidModes.c	2011-12-18 13:44:54.000000000 +0900
+++ xorg-server-1.11.4/hw/xfree86/modes/xf86EdidModes.c	2012-06-05 16:16:02.510844428 +0900
@@ -743,6 +743,10 @@
 	for (j = 7; j > 0; j--) {
 	    if (est[i] & (1 << j)) {
 		m = (i * 8) + (7 - j);
+		if(m >= sizeof(EstIIIModes) / sizeof(EstIIIModes[0]))
+		{
+		    continue;
+		}
 		modes = xf86ModesAdd(modes,
 				     FindDMTMode(EstIIIModes[m].w,
 						 EstIIIModes[m].h,
