diff -x '*~' -urN xf86-input-wacom-0.14.0.orig/src/wcmUSB.c xf86-input-wacom-0.14.0/src/wcmUSB.c
--- xf86-input-wacom-0.14.0.orig/src/wcmUSB.c	2012-03-07 09:27:31.000000000 +0900
+++ xf86-input-wacom-0.14.0/src/wcmUSB.c	2012-06-06 14:15:53.079441043 +0900
@@ -371,7 +371,7 @@
 		 * then they must be associated with pad buttons.
 		 */
 		for (i = ARRAY_SIZE(mouse_codes); i > 0; i--)
-			if (ISBITSET(common->wcmKeys, mouse_codes[i]))
+			if (ISBITSET(common->wcmKeys, mouse_codes[i - 1]))
 				break;
 
 		/* Make sure room for fixed map mouse buttons.  This
